ncm-ceph/src/main/perl/Ceph/OSDserver.pm

```diff
@@ -0,0 +1,173 @@
+#${PMpre} NCM::Component::Ceph::OSDserver${PMpost}
+
+use 5.10.1;
+
+use parent qw(CAF::Object NCM::Component::Ceph::Commands);
+use NCM::Component::Ceph::Cfgfile;
+use EDG::WP4::CCM::Path qw(escape unescape);
+use Readonly;
+use JSON::XS;
+use Data::Dumper;
+
+Readonly my @BOOTSTRAP_OSD_KEYRING_CMD => qw(stat /var/lib/ceph/bootstrap-osd/ceph.keyring);
+Readonly my @BOOTSTRAP_OSD_KEYRING_CMD_SL => qw(stat /etc/ceph/ceph.client.bootstrap-osd.keyring);
+Readonly my @BOOTSTRAP_OSD_CEPH_HEALTH => qw(status --id bootstrap-osd);
+Readonly my @GET_CEPH_PVS_CMD => (qw(pvs -o), 'pv_name,lv_tags', qw(--no-headings --reportformat json));
+
+sub _initialize
+{
+    my ($self, $cfgtree, $log, $prefix) = @_;
+
+    $self->{log} = $log;
+    $self->{cfgtree} = $cfgtree;
+    $self->{prefix} = $prefix;
+    $self->{config} = $cfgtree->getTree($self->{prefix});
+    $self->{ok_failures} = $self->{config}->{daemons}->{max_add_osd_failures};
+
+    $self->{osds} = $self->{config}->{daemons}->{osds};
+    return 1;
+}
+
+sub is_node_healthy
+{
+    my ($self) = @_;
+    # Check bootstrab-osd keyring
+    # stat /var/lib/ceph/bootstrap-osd/ceph.keyring
+    $self->run_command([@BOOTSTRAP_OSD_KEYRING_CMD], "stat bootstrap-osd keyring", test => 1) or return;
+    $self->run_command([@BOOTSTRAP_OSD_KEYRING_CMD_SL], "stat bootstrap-osd keyring symlink", test => 1) or return ;
+    # Checks can be added
+    if (!$self->run_ceph_command([@BOOTSTRAP_OSD_CEPH_HEALTH], "get cluster state")) {
+        $self->error('Cluster not reachable or correctly configured');
+        return;
+    }
+    return 1;
+
+}
+
+sub run_pvs
+{
+    my ($self, $osds) = @_;
+    my ($ok, $jstr) = $self->run_command([@GET_CEPH_PVS_CMD], "get ceph pvs");
+    return if (!$ok);
+    my $report = decode_json($jstr);
+    $self->debug(4, Dumper($report));
+    if (!defined($report->{report}[0]->{pv})) {
+        $self->error('Could not procces pvs json output');
+        return;
+    }
+    my $pvs = $report->{report}[0]->{pv};
+    foreach my $pv (@$pvs) {
+        if ($pv->{lv_tags} =~ m/ceph.osd_id=(\d+),/) {
```

Comment:
why the `,`? you sure there will always be something after? `\d+` is greedy, so you shouldn't worry about that